toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -864,3 +894,419 @@ func createLiveOSIsoImage(buildDir, baseConfigPath string, isoConfig *imagecusto
+
+// createImageFromUnchangedOS
+//
+//   - assuming the LiveOSIsoBuilder instance has all its artifacts populated,
+//     this function goes straight to updating grub and re-packaging the
+//     artifacts into an iso image. It does not re-create the initrd.img or
+//     the squashfs.img. This speeds-up customizing iso images when there are
+//     no customizations applicable to the OS (i.e. to the squashfs.img).
+//
+// inputs:
+//
+//   - 'baseConfigPath':
+//     path to where the configuration is loaded from. This is used to resolve
+//     relative paths.
+//   - 'isoConfig'
+//     user provided configuration for the iso image.
+//   - 'outputImageDir':
+//     path to a folder where the generated iso will be placed.
+//   - 'outputImageBase':
+//     base name of the image to generate. The generated name will be on the
+//     form: {outputImageDir}/{outputImageBase}.iso
+//
+// outputs:
+//
+//   - creates an iso image.
+func (b *LiveOSIsoBuilder) createImageFromUnchangedOS(baseConfigPath string, isoConfig *imagecustomizerapi.Iso, outputImageDir string, outputImageBase string) error {
+
+	logger.Log.Infof("Creating LiveOS iso image using unchanged OS partitions")
+
+	additionalIsoFiles, extraCommandLine, err := micIsoConfigToIsoMakerConfig(baseConfigPath, isoConfig)
+	if err != nil {
+		return fmt.Errorf("failed to convert iso configuration to isomaker configuration format:\n%w", err)
+	}
+
+	err = b.updateGrubCfg(b.artifacts.grubCfgPath, extraCommandLine)
+	if err != nil {
+		return fmt.Errorf("failed to update grub.cfg:\n%w", err)
+	}
+
+	err = b.createIsoImage(additionalIsoFiles, outputImageDir, outputImageBase)
+	if err != nil {
+		return fmt.Errorf("failed to create iso image:\n%w", err)
+	}
+
+	return nil
+}
+
+// getSizeOnDiskInBytes
+//
+//   - given a folder, it calculates the total size in bytes of its contents.
+//
+// inputs:
+//
+//   - 'rootDir':
+//     root folder to calculate its size.
+//
+// outputs:
+//
+//   - returns the size in bytes.
+func getSizeOnDiskInBytes(rootDir string) (size uint64, err error) {
+	logger.Log.Debugf("Estimating disk size for (%s)", rootDir)
+
+	duParams := []string{"-sh", rootDir}
+	duStdout, _, err := shell.Execute("du", duParams...)
+	if err != nil {
+		return 0, fmt.Errorf("failed to find the size of the specified folder (%s):\n%w", rootDir, err)
+	}
+
+	// parse and get count and unit
+	diskSizeRegex := regexp.MustCompile(`^(\d+)([KMGT])?.*`)
+	matches := diskSizeRegex.FindStringSubmatch(duStdout)
+	if matches == nil || len(matches) < 3 {
+		return 0, fmt.Errorf("disk size (%s) has incorrect format. Expecting: ^[0-9]+[KMGT]?\t.* (e.g. 100M, 1G)", duStdout)
+	}
+
+	numString := matches[1]
+	unit := matches[2]
+	num, err := strconv.ParseUint(numString, 0, 10)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse disk size (%s):\n%w", numString, err)
+	}
+	logger.Log.Debugf("du returned=%d %s", num, unit)
+
+	// convert to bytes
+	multiplier := uint64(1)
+	switch unit {
+	case "K":
```

Comment:
If you drop the `-h` to `du`, you won't need to parse the unit.

Reply:
cool! pending.